firmware/Sources/HAL/HAL_ADC/hal_adc.h

```diff
@@ -0,0 +1,106 @@
+/*********************************************************************************
+* @file           : hal_adc.h
+* @brief          : HAL header file for ADC
+***********************************************************************************/
+
+#ifndef HAL_ADC_H_
+#define HAL_ADC_H_
+/**********************************************************************************/
+/*                               Project Includes                                 */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                              Include other headers                             */
+/**********************************************************************************/
+#include <stdint.h>
+
+/**********************************************************************************/
+/*                     Definition of local symbolic constants                     */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                    Definition of local function like macros                    */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*            Definition of exported types (typedef, enum, struct, union)         */
+/**********************************************************************************/
+
+/**
+ * @enum HAL_ADC_result_e
+ * @brief Structure for the result of the ADC operation.
+ */
+typedef enum
+{
+	HAL_ADC_RESULT_SUCCESS 	= 0x00U, 	/**< HAL_ADC success operation result **/
+	HAL_ADC_RESULT_ERROR	 	= 0x01U,	/**< HAL_ADC error operation result **/
+	HAL_ADC_RESULT_BUSY			= 0x02U,	/**< HAL_ADC busy operation result **/
+	HAL_ADC_RESULT_TIMEOUT	= 0x03U,	/**< HAL_ADC timeout operation result **/
+} HAL_ADC_result_e;
+
+
+/**
+ * @enum HAL_ADC_port_e
+ * @brief ADC channels.
+ */
+typedef enum
+{
+	HAL_ADC_LS_CURR,			/** ADC2_IN1 - PA4 **/
+	HAL_ADC_LS_VOLT,			/** ADC2_IN2 - PA5 **/
+	HAL_ADC_HS_VOLT,			/** ADC2_IN4 - PA6 **/
+
+	HAL_ADC_EXT_TEMP1,		/** ADC1_IN1 - PA5 **/
```

Comment:
3 temps?
Should you asign a more redeable name?

Reply:
Done